### PR TITLE
Backport: pyroscope.ebpf: Update pyroscope/ebpf to v0.4.6

### PR DESCRIPTION
**Pull request by simonswine**
This backports into Grafana Agent [ (#678)](https://github.com/grafana/alloy/pull/678)

### CODE REVIEW

**Review by clayton-cornell — APPROVED** [review]
Docs look OK. Needs a review from @grafana/grafana-agent-maintainers

**Review by korniltsev — APPROVED** [review]

**Comment by korniltsev** [comment]
I am surprised we do changes in agent. Will there be new agent releases?

**Comment by simonswine** [comment]
@korniltsev I think so, there is a [internal document](https://docs.google.com/document/d/1jzgRYMTAY80wMz56zJYiPgnd4ZcnHMRVUqI4jGrBLrM/edit#heading=h.zc0r8k7iniqa) on this

**Pull request merged**
